system/cpuinfo.c

```diff
@@ -90,6 +90,13 @@ static void determine_cache_size()
         }
         // Zhaoxin CPU only
         /* fall through */
+      case 'V':
+        // Vortex86
+        if (cpuid_info.version.family < 6) {
```

Comment:
'fall through' are dangerous and shouldn't be stacked if possible. What if Zhaoxin release a CPU with family < 6 ?

Reply:
I've added an additional check to ensure that the fall through only handles Vortex86 CPUs.
Actually, looking at the CPUID dumps from Zhaoxin family 7 CPUs, it has cache info in both Intel-style and AMD-style extended information so that could be simplified though perhaps for another PR (some might have a different vendor ID string as well that starts with a space ` `).

Reply:
There is also a potential issue with VIA Nano CPUs without padlock (which returns "VIA VIA VIA " instead of the usual "CentaurHauls"). I will check the cpuz database if this may fail somewhere.

CPU detection is always more complex than initially expected 😵‍💫

Reply:
Finally, I didn't find any of these CPU in the CPU-Z database (with cpuid "VIA VIA VIA ", so I think it's safe to merge as-is.